cranelift-codegen/src/context.rs

```diff
@@ -132,33 +132,37 @@ impl Context {
         self.verify_if(isa)?;
         debug!("Compiling:\n{}", self.func.display(isa));
 
+        let opt_level = isa.flags().opt_level();
+
         self.compute_cfg();
-        if isa.flags().opt_level() != OptLevel::Fastest {
+        if opt_level != OptLevel::Fastest {
             self.preopt(isa)?;
         }
         if isa.flags().enable_nan_canonicalization() {
             self.canonicalize_nans(isa)?;
         }
         self.legalize(isa)?;
-        if isa.flags().opt_level() != OptLevel::Fastest {
+        if opt_level != OptLevel::Fastest {
             self.postopt(isa)?;
         }
-        if isa.flags().opt_level() == OptLevel::Best {
+        if opt_level == OptLevel::NearlyBest || opt_level == OptLevel::Best {
             self.compute_domtree();
             self.compute_loop_analysis();
             self.licm(isa)?;
             self.simple_gvn(isa)?;
         }
         self.compute_domtree();
         self.eliminate_unreachable_code(isa)?;
-        if isa.flags().opt_level() != OptLevel::Fastest {
+        if opt_level != OptLevel::Fastest {
             self.dce(isa)?;
         }
         self.regalloc(isa)?;
         self.prologue_epilogue(isa)?;
-        if isa.flags().opt_level() == OptLevel::Best {
+        if opt_level == OptLevel::NearlyBest || opt_level == OptLevel::Best {
             self.redundant_reload_remover(isa)?;
-            self.shrink_instructions(isa)?;
+            if opt_level == OptLevel::Best {
```

Comment:
Maybe move this if out of the if above. Also performing this is necessary for gdb to show correct backtraces. (https://github.com/bjorn3/rustc_codegen_cranelift/issues/146#issuecomment-449474527)